main.tf

```diff
@@ -262,3 +266,44 @@ data "ibm_database_connection" "database_connection" {
   user_id       = ibm_database.elasticsearch.adminuser
   user_type     = "database"
 }
+
+##############################################################################
+# ELSER support
+##############################################################################
+
+locals {
```

Comment:
could you add some supporting code comments to explain whats going on in the logic below?